interfaces/repo.go

```diff
@@ -576,19 +570,54 @@ func (r *Repository) Connect(ref *ConnRef, plugDynamicAttrs, slotDynamicAttrs ma
 	// Ensure that such plug exists
 	plug := r.plugs[plugSnapName][plugName]
 	if plug == nil {
-		return nil, &UnknownPlugSlotError{Msg: fmt.Sprintf("cannot connect plug %q from snap %q: no such plug", plugName, plugSnapName)}
+		return nil, nil, &UnknownPlugSlotError{Msg: fmt.Sprintf("cannot connect plug %q from snap %q: no such plug", plugName, plugSnapName)}
 	}
 	// Ensure that such slot exists
 	slot := r.slots[slotSnapName][slotName]
 	if slot == nil {
-		return nil, &UnknownPlugSlotError{fmt.Sprintf("cannot connect slot %q from snap %q: no such slot", slotName, slotSnapName)}
+		return nil, nil, &UnknownPlugSlotError{fmt.Sprintf("cannot connect slot %q from snap %q: no such slot", slotName, slotSnapName)}
 	}
 	// Ensure that plug and slot are compatible
 	if slot.Interface != plug.Interface {
-		return nil, fmt.Errorf(`cannot connect plug "%s:%s" (interface %q) to "%s:%s" (interface %q)`,
+		return nil, nil, fmt.Errorf(`cannot connect plug "%s:%s" (interface %q) to "%s:%s" (interface %q)`,
 			plugSnapName, plugName, plug.Interface, slotSnapName, slotName, slot.Interface)
 	}
 
+	return plug, slot, nil
+}
+
+// Connect re-establishes a connection between a plug and a slot as part of connection reloading.
+// When connections are reloaded policyCheck is null (we don't check policy again).
+func (r *Repository) ConnectReload(ref *ConnRef, plugStaticAttrs, plugDynamicAttrs, slotStaticAttrs, slotDynamicAttrs map[string]interface{}) (*Connection, error) {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	plug, slot, err := r.getInfos(ref)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, ok := r.ifaces[plug.Interface]; !ok {
+		return nil, fmt.Errorf("internal error: unknown interface %q", plug.Interface)
+	}
+
+	cplug := NewConnectedPlugReload(plug, plugStaticAttrs, plugDynamicAttrs)
+	cslot := NewConnectedSlotReload(slot, slotStaticAttrs, slotDynamicAttrs)
```

Comment:
That also seems a bit curious in terms of API. Just reading it out loud seems to make no sense.. It's a new connected plug.. reload?

Seems better to leave the original functions alone and then tuning the results as necessary.

Reply:
I agree, that name is not very fortunate as well...
As for the suggestion about tuning the results - the problem is that ConnectedPlug/Slot types deliberately do not allow modifications of static attributes, therefore this new constructor here.